### ISSUE TITLE
Define consistent mini library exports with base classes included

### ISSUE THREAD

**Issue by navibyte** [navibyte/geospatial#31]
Some mini libraries should also export some of base classes.

For example if needing GeoJSON parser but also base geometries, you have to import:
```dart
import 'package:geocore/base.dart';
import 'package:geocore/parse_geojson.dart';
```

Maybe it would be handy if import for `parse_geojson.dart` would export also base geometries (and factory abstractions).

## Libraries (currently in version 0.62 situation) for geocore

The package contains following mini-libraries:

Library              | Description 
-------------------- | -----------
**base**             | Geometry classes including points, bounds, line strings, polygons and more.
**crs**              | Classes to represent coordinate reference system (CRS) identifiers.
**feature**          | Feature and FeatureCollection to handle dynamic geospatial data objects.
**geo**              | Geographic points and bounds classes to represent longitude-latitude data
**meta_extent**      | Metadata structures to handle extents with spatial and temporal parts.
**parse_factory**    | Base interfaces and implementations for geospatial data factories.
**parse_geojson**    | Geospatial data factory to parse GeoJSON from text strings.
**parse_wkt**        | Geospatial data factory to parse WKT from text strings.

For example to access a mini library you should use an import like:

```dart
import 'package:geocore/base.dart';
```

To use all libraries of the package:

```dart
import 'package:geocore/geocore.dart';
```

**Comment by navispatial** [comment]
_Latest plans for restructuring, coming soon_:

The package contains also following mini-libraries, that can be used to import
only a certain subset instead of the whole **geocore** library:

Library         | Export also | Description 
--------------- | ----------- | -----------------------------------------------
**base**        | | Base classes for coordinates, points, bounds, point series and transforms.
**common**      | | Common data structures (like temporal instants and intervals).
**coordinates** | base | Cartesian and geographic points with some common coordinate transforms.
**data**        | base, common, coordinates | Geospatial features and geometries (linestring, polygon, multi geometries).
**parse**       | base, common, coordinates, data | GeoJSON and WKT (Well-known text representation of geometry) data parsers.
**proj4d**      | base | Projections provided by the external [proj4dart](https://pub.dev/packages/proj4dart) package.

All the mini-libraries have a dependency to the 
[equatable](https://pub.dev/packages/equatable) package. The **proj4d** library 
depends also on the [proj4dart](https://pub.dev/packages/proj4dart) package.

**Comment by navispatial** [comment]
Implemented in `Pre-release 0.8.0-a.5`.
